### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by dmitraver**
Hey! I've updated the akka version to 2.4. Also I changed initial replicated log index to 1 because now our logs are 1-indexed and changed few tests as well. There are still some of them failing but I'm looking into it.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1 +1 @@
-crossScalaVersions := Seq("2.11.4", "2.10.4")
\ No newline at end of file
+scalaVersion := "2.11.7"
```
Comment (ktoso) [inline build.sbt]: Good, thanks!

File: project/Build.scala [hunk 2]
```diff
@@ -36,13 +36,13 @@ object ApplicationBuild extends Build {
 }
 
 object Dependencies {
-    val akkaVersion = "2.3.8"
+    val akkaVersion = "2.4.0"
```
Comment (ktoso) [inline project/Build.scala]: great :)

File: src/main/scala/pl/project13/scala/akka/raft/Follower.scala [hunk 3]
```diff
@@ -55,11 +55,16 @@ private[raft] trait Follower {
         stay() using m.withVoteFor(candidate)
       }
 
-    case Event(RequestVote(term, candidateId, lastLogTerm, lastLogIndex), m: Meta) =>
+    case Event(RequestVote(term, candidateId, lastLogTerm, lastLogIndex), m: Meta) if m.votedFor.isDefined =>
```
Comment (ktoso) [inline src/main/scala/pl/project13/scala/akka/raft/Follower.scala]: thanks!

File: src/main/scala/pl/project13/scala/akka/raft/model/ReplicatedLog.scala [hunk 4]
```diff
@@ -24,7 +24,7 @@ case class ReplicatedLog[Command](
 
   // log state
   def lastTerm  = entries.lastOption map { _.term } getOrElse Term(0)
-  def lastIndex = entries.lastOption map { _.index } getOrElse 0
+  def lastIndex = entries.lastOption map { _.index } getOrElse 1
```
Comment (ktoso) [inline src/main/scala/pl/project13/scala/akka/raft/model/ReplicatedLog.scala]: we <3 1-off errors, don't we all? Thanks!

**Comment by ktoso** [comment]
Thanks for the fixes, looks good.
Thanks also in general for the help!

**Comment by dmitraver** [comment]
@ktoso I enjoy working on it ;)

**Comment by ktoso** [comment]
:-)

**Pull request merged**